sdk/lib/opentelemetry/sdk/trace/trace_state.rb

```diff
@@ -0,0 +1,186 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Trace
+      # TraceState represents the OpenTelemetry::Trace::Tracestate values that
+      # are part of the OpenTelemetry ecosystem, contained in a single entry
+      # using the `ot` key. The format and handling requirements are specified
+      # at https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/tracestate-handling.md
+      class TraceState
+        class << self
+          private :new # rubocop:disable Style/AccessModifierDeclarations
+
+          # Returns a newly created TraceState parsed from the Tracestate provided.
+          #
+          # @param [OpenTelemetry::Trace::Tracestate] tracestate
+          # @return [TraceState] A new TraceState instance or DEFAULT
+          def from_tracestate(tracestate) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+            return DEFAULT if tracestate.empty?
+
+            ot = tracestate.value('ot')
+            return DEFAULT if ot.nil? || ot.length > MAX_LIST_LENGTH
+
+            hash = ot.split(';').each_with_object({}) do |member, memo|
+              member.strip!
```

Comment:
We could have spaces in a W3C trace context value (apparently, from reading the spec), but nothing in the `ot` key should ever have a space ... and yet I know it's wise to be defensive here.

I have no useful comment, I just hate computers and specs sometimes.